### PR TITLE
Implement Animal Crossing: Happy Home Designer DataStore protocol

### LINKED ISSUE #76: [Feature]: Implement the Data Store (Animal Crossing: Happy Home Designer) Protocol
### Checked Existing

- [X] I have checked the repository for duplicate issues.

### What feature do you want to see added?

The Animal Crossing: Happy Home Designer's [additional methods in the Data Store protocol](https://nintendo-wiki.pretendo.network/docs/nex/protocols/datastore/happy-home-designer).

### Why do you want to have this feature?

Happy Home Designer's online services appear to be very similar to Super Mario Maker's, even for a gameplay style. I have a feeling that it was a test to see if online data sharing could be done on a 3DS, and then the same server and codebase was used in the Super Mario Maker port to the 3DS. The UI even looks the same - and so are the method names in the extended Data Store protocol.

### Any other details to share? (OPTIONAL)

_No response_

### PR DESCRIPTION
**Pull request by ItzSwirlz**
Resolves #76

### Changes:
- Implements Animal Crossing: Happy Home Designer's additional methods to the Data Store protocol.

- [X] I have read and agreed to the [Code of Conduct](https://github.com/PretendoNetwork/Pretendo/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have read and complied with the [contributing guidelines](https://github.com/PretendoNetwork/Pretendo/blob/master/.github/CONTRIBUTING.md).
- [ ] What I'm implementing was an [approved issue](../issues?q=is%3Aopen+is%3Aissue+label%3Aapproved).
- [ ] I have tested all of my changes.

### CODE REVIEW

**Comment by ItzSwirlz** [comment]
Phew! All done!

File: datastore/ac-happy-home-designer/protocol.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	ProtocolID = 0x73
+
+	// MethodGetObjectInfos is the method ID for GetObjectInfos
+	MethodGetObjectInfos = 0x2D
+
+	// MethodGetMetaByOwnerId is the method ID for GetMetaByOwnerId
+	MethodGetMetaByOwnerId = 0x2E
+
+	// MethodGetMetaByUniqueId is the method ID for GetMetaByUniqueId
+	MethodGetMetaByUniqueId = 0x2F
```
Comment (DaniElectra) [inline datastore/ac-happy-home-designer/protocol.go:26]:
```suggestion
	// MethodGetMetaByOwnerID is the method ID for GetMetaByOwnerID
	MethodGetMetaByOwnerID = 0x2E

	// MethodGetMetaByUniqueID is the method ID for GetMetaByUniqueID
	MethodGetMetaByUniqueID = 0x2F
```

Same for every ocurrence

File: datastore/ac-happy-home-designer/types/data_store_get_meta_by_owner_id_param.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+// DataStoreGetMetaByOwnerIdParam is a type within the DataStoreACHappyHomeDesigner protocol
+type DataStoreGetMetaByOwnerIdParam struct {
+	types.Structure
+
+	OwnerIds     *types.List[*types.PrimitiveU32]
+	DataTypes    *types.List[*types.PrimitiveU16]
+	ResultOption *types.PrimitiveU8
+	ResultRange  *types.ResultRange
+}
```
Comment (DaniElectra) [inline datastore/ac-happy-home-designer/types/data_store_get_meta_by_owner_id_param.go:19]:
```suggestion
// DataStoreGetMetaByOwnerIDParam is a type within the DataStoreACHappyHomeDesigner protocol
type DataStoreGetMetaByOwnerIDParam struct {
	types.Structure
	OwnerIDs     *types.List[*types.PrimitiveU32]
	DataTypes    *types.List[*types.PrimitiveU16]
	ResultOption *types.PrimitiveU8
	ResultRange  *types.ResultRange
}
```

File: datastore/ac-happy-home-designer/types/data_store_get_meta_by_unique_id_param.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+// DataStoreGetMetaByUniqueIdParam is a type within the DataStoreACHappyHomeDesigner protocol
+type DataStoreGetMetaByUniqueIdParam struct {
+	types.Structure
+
+	UniqueIds    *types.List[*types.PrimitiveU32]
+	DataTypes    *types.List[*types.PrimitiveU16]
+	ResultOption *types.PrimitiveU8
+	ResultRange  *types.ResultRange
+}
```
Comment (DaniElectra) [inline datastore/ac-happy-home-designer/types/data_store_get_meta_by_unique_id_param.go:19]:
```suggestion
// DataStoreGetMetaByUniqueIDParam is a type within the DataStoreACHappyHomeDesigner protocol
type DataStoreGetMetaByUniqueIDParam struct {
	types.Structure
	UniqueIDs    *types.List[*types.PrimitiveU32]
	DataTypes    *types.List[*types.PrimitiveU16]
	ResultOption *types.PrimitiveU8
	ResultRange  *types.ResultRange
}
```

**Comment by jonbarrow** [comment]
As much as we appreicate you taking the time to implement these things, I ***have*** mentioned already that none of this work you're doing is going to be used. We no longer wish to have game-specific protocols in this library, opting to implement them on a per-server basis, and removing them from here as that happens. Which means none of this will ever actually see the light of day.

I know this was a lot of work, but this is the exact reason why the "approved issues" system was put in place. To avoid people doing work which we won't use/aren't interested in.

**Comment by ItzSwirlz** [comment]
Oh, ok... whoops. Well, good practice I guess.

**Pull request closed without merging**